wordpress-mobile/WordPress-Android
# Implement edit category feature for category management

Parent issue - https://github.com/wordpress-mobile/WordPress-Android/issues/16058
Closes #16087

This PR adds the Update category functionality to the app 

Update category Screen 

![Screenshot_20221206_153534](https://user-images.githubusercontent.com/17463767/206232909-d14ba9d3-3437-4091-9632-54c0834fa502.png)


## To test:

Prerequisites
- Go to My Site -> Me -> App Settings -> Test Feature Configuration
- Enable ManageCategoriesFeatureConfig and restart the app
- Relaunch the app.

### Test 1
- Launch app with a site having Categories
- Go to Categories Screen:  My Site -> Site Settings -> Categories
- Click at any categories
- Verify that the correct Category is selected in Parent Category Spinner
- Verify that the title and button text is **Update Category**
- Verify that only on Typing in the category name, the`Update Category` button gets enabled 
- On Clicking `Update Category` button, progress dialog is shown 
- Verify that On Success, Toast with message `Category updated successfully` is shown 
- Verify that the User is redirected to the Categories Listing screen with the Updated category present 

### Test 2 - Updating Category Failed 
- Follow the steps 1-4 as in Test 1
- Try to update the category name with an already existing category name  
- Notice that a Toast with the message `Updating category failed` is shown 

## Regression Notes
1. Potential unintended areas of impact
Adding and Updating category doesn't work as intended

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and Unit tests 

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
